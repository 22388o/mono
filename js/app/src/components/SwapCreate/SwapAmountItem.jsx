import React, { useCallback, useSyncExternalStore } from 'react'
import styles from '../../styles/SwapCreate.module.css'
import { Grid, Button } from '@mui/material'
import { walletStore } from '../../syncstore/walletstore'

export const SwapAmountItem = ({ assetId, amount, className, onAmountChange, unitPrice, availQty }) => {
  const globalWallet = useSyncExternalStore(walletStore.subscribe, () => walletStore.currentState)
  const asset = globalWallet.assets[assetId]

  const onKeyDown = useCallback((e) => {
    if (e.keyCode === 109) {
      e.preventDefault()
    }
  }, [])

  return (
    <>
      <Grid container direction='row' className={className} sx={{ height:'48px' }}>
        <Grid item xs={7} container direction='column' textAlign='left'>
          <input
            className={`${styles['qty-input']} qty-input`}
            placeholder={availQty}
            type='number'
            value={(amount === 0) ? '' : amount}
            onChange={onAmountChange}
            onKeyDown={onKeyDown}
          />
          {unitPrice * amount > 0 ? <span className={styles.prices}>${unitPrice * amount}</span> : ''}
        </Grid>
        <Grid item xs={5} textAlign='right' sx={{display:'flex',alignItems:'center',justifyContent:'flex-end'}}>
          <Button className={`${styles['coin-select']} coin-select`} onClick={() => {}}><img src={asset.img_url} />{asset.type.split('-')[0]}</Button>
        </Grid>
      </Grid>
    </>
  )
}
