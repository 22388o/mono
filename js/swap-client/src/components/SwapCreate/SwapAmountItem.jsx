import React from 'react';
import { Dropdown, Form } from 'semantic-ui-react';
import styles from '../styles/SwapCreate.module.css';

const friendOptions = [
  {
    key: 'BTC',
    text: 'BTC',
    value: 'BTC',
    image: { avatar: true, src: 'https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true' },
  },
  {
    key: 'ETH',
    text: 'ETH',
    value: 'ETH',
    image: { avatar: true, src: 'https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true' },
  }
]

export const SwapAmountItem = ({coinType, amount, className, onAmountChange, unitPrice, onCoinTypeChange, availQty, limitOrder = true}) => {
  const links = {
    'btc': "https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true",
    'eth': "https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true"
  };

  const onKeyDown = (e) => {
    if(e.keyCode === 109) {
      e.preventDefault();
      return;
    }
  }

  return <Form.Group widths='equal' className={className}>
    <Form.Field className={styles.swapAmountInput}>
      { limitOrder ? 
          <input placeholder={availQty} type='number' value={(amount===0)?'':amount} onChange={onAmountChange} onKeyDown={onKeyDown}/>
        :
          <input placeholder={availQty} type='number' value={(amount===0)?'':amount} onChange={onAmountChange} onKeyDown={onKeyDown}/>
      }
      { unitPrice * amount > 0 ? <p className={styles.price}>${unitPrice * amount}</p> : ''}
    </Form.Field>
    <Form.Field className={styles.coinType}>
      <Dropdown
        className={styles.swapCoinSelect}
        floating
        fluid
        labeled
        button
        value={coinType}
        options={friendOptions}
        onChange={onCoinTypeChange}
      />
    </Form.Field>
  </Form.Group>
};