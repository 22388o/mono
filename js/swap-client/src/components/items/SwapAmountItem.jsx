import React from 'react';
import { Form } from 'semantic-ui-react';
import styles from '../styles/SwapCreate.module.css';

export const SwapAmountItem = ({coinType, amount, className, onChange}) => {
  const links = {
    'BTC': "https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true",
    'ETH': "https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/ethereum/info/logo.png?raw=true"
  };
  return <Form.Group widths='equal' className={className}>
    <Form.Field>
      <input className={styles.swapInput} type='number' value={amount} onChange={onChange}/>
    </Form.Field>
    <Form.Field className={styles.coinType}>
      <div className='ui inline dropdown'>
        <div className={styles.SwapFormText}>
          <img className="ui avatar image" src={links[coinType]} />
          {coinType.toUpperCase()}
        </div>
        {/* <i className="dropdown icon"></i> */}
        <div className="menu">
          <div className="item">
            <img className="ui avatar image" src={links[coinType]} />
            BTC
          </div>
          <div className="item">
            <img className="ui avatar image" src={links[coinType]} />
            ETH
          </div>
        </div>
      </div>
    </Form.Field>
  </Form.Group>
};