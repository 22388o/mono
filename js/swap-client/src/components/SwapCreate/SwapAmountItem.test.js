import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // For the "toBeInTheDocument" matcher
import { SwapAmountItem } from './SwapAmountItem';

describe('CollectiblesModal component test', () => {
  it('renders coin component', () => {

    const onCoinTypeChange = jest.fn();

    const {container, getByText} = render(
      //coin-select
      <SwapAmountItem
        assetId={0}
        unitPrice={28000}
        amount={1000}
        availQty={1000}
        onAmountChange={() => {}}
        onCoinTypeChange={onCoinTypeChange}
      />
    );

    const coinSelectElement = container.querySelector('.coin-select');
    fireEvent.click(coinSelectElement);

    const modalHeader = getByText('Select Asset');

    expect(container).toMatchSnapshot();
    expect(modalHeader).toBeInTheDocument(); //when cointype select button is clicked, check if modal is opened
    
    const btcBtn = getByText('Bitcoin');
    fireEvent.click(btcBtn);

    expect(onCoinTypeChange).toHaveBeenCalled(); // when coin type is changed, check if the function is called
  });
});
