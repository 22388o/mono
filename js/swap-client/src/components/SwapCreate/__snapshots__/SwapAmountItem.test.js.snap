// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectiblesModal component test renders coin component 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiGrid-root MuiGrid-container css-1tj2tdk-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-7 css-1smoqvn-MuiGrid-root"
    >
      <input
        class="undefined qty-input"
        placeholder="1000"
        type="number"
        value="1000"
      />
      <span>
        $
        28000000
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-14m1l2r-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium undefined coin-select css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <img
          src="https://github.com/dapphub/trustwallet-assets/blob/master/blockchains/bitcoin/info/logo.png?raw=true"
        />
        BTC
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`CollectiblesModal component test renders ordinal component 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiGrid-root MuiGrid-container css-1tj2tdk-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-7 css-1ggbfv-MuiGrid-root"
    >
      <img
        src="/nft/2.png"
      />
      <div
        class="MuiStack-root css-cuozto-MuiStack-root"
      >
        <span>
          Inscription
        </span>
        <span>
          2c0
          ...
          ei0
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-14m1l2r-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium undefined coin-select css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <img
          src="/nft/2.png"
        />
        BTCORD
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
